source/common/network/address_impl.h

```diff
@@ -53,6 +69,21 @@ class InstanceBase : public Instance {
   const Type type_;
 };
 
+class InstanceFactory {
```

Comment:
Since this is used in multiple `.cc`, it's basically a public interface, can you please document it better? Thanks.